apiclient/client.py

```diff
@@ -21,6 +21,7 @@ def __init__(
         response_handler: Type[BaseResponseHandler] = RequestsResponseHandler,
         request_formatter: Type[BaseRequestFormatter] = NoOpRequestFormatter,
         error_handler: Type[BaseErrorHandler] = ErrorHandler,
+        request_strategy: Type[BaseRequestStrategy] = RequestStrategy,
```

Comment:
```suggestion
        request_strategy: BaseRequestStrategy = RequestStrategy(),
```

Reply:
That's not compatible behavior with the current impl though because of the "function defaults evaluated at compile time" thing and that the strategy stores a copy of the client on itself. E.g.,:

This is current behavior:

```python
import apiclient
c1 = apiclient.APIClient()
c2 = apiclient.APIClient()
c1._request_strategy is c2._request_strategy  # False
c1._request_strategy._client is c2._request_strategy._client  # False
```

But if the default is instantiated in the function signature:

```python
import apiclient
c1 = apiclient.APIClient()
c2 = apiclient.APIClient()
c1._request_strategy is c2._request_strategy  # True
c1._request_strategy._client is c2._request_strategy._client  # True
```

Would it be better if we did this?

```python
   def __init__(
        self,
        ...,
        request_strategy: Optional[BaseRequestStrategy] = None,
    ):
        ...
        if request_strategy is None:
            request_strategy = RequestStrategy()
        self.set_request_strategy(request_strategy)
```

Reply:
Okay, that's a fair point.  Your proposed `Optional` does work better.